cert-manager's Gateway API support is no longer experimental in cert-manager 1.15

maelvls opened the pull request:
Hey,

I'm happy to report that cert-manager has graduated the Gateway API support to beta! You no longer have to pass the feature flag `ExperimentalGatewayAPISupport` anymore. Most importantly, the feature flag `ExperimentalGatewayAPISupport` was removed, so folks who used it in cert-manager 1.14 and below will need to remove it and add the new flag `--enable-gateway-api`.

Source: https://cert-manager.io/docs/releases/upgrading/upgrading-1.14-1.15#gatewayapi-promotion-to-beta

Ref: https://github.com/cert-manager/website/issues/851#issuecomment-2230387300

```release-note
NONE
```

maelvls committed:
gateway api no longer experimental in cert-manager 1.15

Signed-off-by: Maël Valais <mael@vls.dev>

sayboras commented on Documentation/network/servicemesh/tls-cert.rst:
```diff
@@ -36,7 +36,7 @@ Create TLS Certificate and Private Key
                 --set installCRDs=true \
                 --create-namespace \
-                --set "extraArgs={--feature-gates=ExperimentalGatewayAPISupport=true}"
+                --set "extraArgs={--enable-gateway-api}"
```
do we need to update the version at line 35 (i.e.  v1.10.0 to v1.15.0) ?

maelvls replied:
Good point, I've updated the version. I've also changed `installCRDs` to `crds.enabled` since `installCRDs` has been deprecated in cert-manager 1.15.

sayboras replied:
Thanks a lot for your kind help :100: :+1: :medal_military:

maelvls committed:
cert-manager: use 1.15.1 instead of 1.11.0

I also changed "installCRDs" to "crds.enabled" since installCRDs has
been deprecated in cert-manager 1.15.

Signed-off-by: Maël Valais <mael@vls.dev>

sayboras reviewed (approved):
LGTM and thanks a lot

mhofstetter commented on Documentation/network/servicemesh/tls-cert.rst:
```diff
@@ -32,11 +32,11 @@ Create TLS Certificate and Private Key
 
             $ helm repo add jetstack https://charts.jetstack.io
-            $ helm install cert-manager jetstack/cert-manager --version v1.10.0 \
+            $ helm install cert-manager jetstack/cert-manager --version v1.15.1 \
```
it seems as this command is failing with the following error whereas the previous command worked without an issue.

```
❯ helm install cert-manager jetstack/cert-manager --version v1.15.1 \
                --namespace cert-manager \
                --create-namespace \
                --set crds.enabled=true \
                --set config.enableGatewayAPI=true
Error: INSTALLATION FAILED: execution error at (cert-manager/templates/controller-config.yaml:2:38): .Values.config.apiVersion must be set !
```

It looks like we need wait for https://github.com/cert-manager/cert-manager/pull/7126 being released or add the missing helm values too.

```
--set config.apiVersion="controller.config.cert-manager.io/v1alpha1" \
--set config.kind="ControllerConfiguration" \
```

edit: a quick test with the missing helm values starts the Helm installation - but it doesn't finish, as the `startupapicheck` pod doesn't succeed. :thinking: not sure whether i miss something in my setup though (local kind cluster with Cilium installed)

```
I0716 15:18:10.341850       1 api.go:108] "Not ready" logger="cert-manager.startupapicheck.checkAPI" err="Internal error occurred: failed calling webhook \"webhook.cert-manager.io\": failed to call webhook: Post \"https://cert-manager-webhook.cert-manager.svc:443/validate?timeout=30s\": tls: failed to verify certificate: x509: certificate signed by unknown authority"
```

learnitall replied:
Thank you for testing out the changes @mhofstetter, I appreciate the attention to detail there.

mhofstetter reviewed (changes requested):
Thanks for the contribution! :+1:  Looks like we still need to improve the command a little bit - see my comment inline.
